x-pack/plugins/dataset_quality/public/services/data_streams_stats/data_streams_stats_client.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { find, merge } from 'lodash';
+import { HttpStart } from '@kbn/core/public';
+import { DataStreamStat } from '../../../common/data_streams_stats/data_stream_stat';
+import { DATA_STREAMS_STATS_URL } from '../../../common/constants';
+import {
+  GetDataStreamsStatsError,
+  GetDataStreamsStatsResponse,
+  GetDataStreamsStatsQuery,
+  DataStreamStatServiceResponse,
+} from '../../../common/data_streams_stats';
+import { IDataStreamsStatsClient } from './types';
+
+export class DataStreamsStatsClient implements IDataStreamsStatsClient {
+  constructor(private readonly http: HttpStart) {}
+
+  public async getDataStreamsStats(
+    params: GetDataStreamsStatsQuery = { type: 'logs' }
+  ): Promise<DataStreamStatServiceResponse> {
+    const { dataStreamsStats, integrations } = await this.http
+      .get<GetDataStreamsStatsResponse>(DATA_STREAMS_STATS_URL, {
+        query: params,
+      })
+      .catch((error) => {
+        throw new GetDataStreamsStatsError(`Failed to fetch data streams stats": ${error}`);
+      });
+
+    const mergedDataStreamsStats = dataStreamsStats.map((statsItem) => {
+      const integration = find(integrations, { name: statsItem.integration });
+
+      return integration ? merge({}, statsItem, { integration }) : statsItem;
+    });
+
+    return mergedDataStreamsStats.map(DataStreamStat.create);
+  }
+}
```

Comment:
**doubt:** Don't we have any encoding/decoding step on this data fetching layer? I don't know the specs for this API, but it feels risky to trust the type only at compile time given the fact this data comes from an ES query.

Reply:
We are using typed routing `kbn-server-route-repository` package which takes care of runtime validations

Reply:
I noticed this is used when registering new endpoints in the `registerRoutes` function for validating the request params/body, what I meant is that I couldn't find any enconding/decoding in the server -> client communication for the response. Is it present somewhere and I'm missing it?

Reply:
Hey @tonyghiani I was checking this in detail, you are right we are not decoding in the client and just relying in the typings there, since we are the ones creating the response and not relying on third party APIs do you think would still add value to have this mechanism in the UI? If so, I can add it in a following PR.

Reply:
Hey Yngrid, thanks for getting back to this merged work to see the comment!
I believe that validating this communication will help us catch any change in schemas or possible edge cases and avoid client failures at runtime, I'm writing more about this in our team guideline, we can keep on this conversation offline, don't worry about updating this work now :)